File: 2696-minimum-string-length-after-removing-substrings/2696-minimum-string-length-after-removing-substrings.py
class Solution(object):
    def minLength(self, s):
        stack = []
        
        for i in s:
            if not stack:
                stack.append(i)
            elif (i == "B" and stack[-1] == "A") or (i == "D" and stack[-1] == "C"):
                stack.pop()
            else:
                stack.append(i)
                
        return len(stack)

File: 2696-minimum-string-length-after-removing-substrings/2696-minimum-string-length-after-removing-substrings.java
class Solution {
    public int minLength(String s) {
        ArrayList<Character> arr = new ArrayList();
        
        for (int i=0; i<s.length(); i++){
            if (arr.isEmpty()){
                arr.add(s.charAt(i));
            }
            else {
                char top = arr.get(arr.size() - 1);
                char curr = s.charAt(i);
                
                if ((Character.toUpperCase(top) == 'A' && Character.toUpperCase(curr) == 'B') ||
                    (Character.toUpperCase(top) == 'C' && Character.toUpperCase(curr) == 'D')){
                    arr.remove(arr.size() - 1);
                }
                else {
                    arr.add(s.charAt(i));
                }
            }
        }
        return arr.size();
    }
}